frontend/src/constants/global.ts

```diff
@@ -1,7 +1,32 @@
+import { formatISO, parseISO, isValid } from 'date-fns';
+
 export const URL = 'http://localhost:8080';
 export const FULLTIME_HOURS = 37.5;
 
 export const WINDOW_WIDTH_THRESHOLD: number = 780;
 
 // A placeholder for authentication token
 export const getAuthenticationToken = (): string => 'xx';
+
+enum DateStrBrand {}
+
+export type DateStr = string & DateStrBrand;
+
+function checkValidDateStr(str: string): str is DateStr {
+  return isValid(parseISO(str));
+}
+
+export function toDateStr(date: Date | string): DateStr {
```

Comment:
I made a design choice to have a strongly typed date string. I don't know if this is good or bad but it follows this https://spin.atomicobject.com/2017/06/19/strongly-typed-date-string-typescript/

Im assuming at somepoint the dates will be returned by the server as strings.

Reply:
The server will return strings inside JSON, but that can be handled by a separate function (e.g. `parseISO`).
I don't understand the objection to using `Date` objects just because they have time as well. We have a library that makes conversions and manipulations easy, I see little reason not to use `Date` everywhere. Maybe it's just me being lazy :stuck_out_tongue: 
But yeah, a second opinion would be nice.